exercises/src/test/scala/fpinscala/LazinessSpec.scala

```diff
@@ -0,0 +1,153 @@
+package fpinscala
+
+import fpinscala.laziness.{ Stream, Empty, Cons}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class LazinessSpec extends AnyFlatSpec with Matchers {
+
+
+  it should "convert a stream to a list" in {
+    Stream(1, 2, 3, 4).toList() shouldBe List(1, 2, 3, 4)
+    Stream().toList() shouldBe Nil
+  }
+
+  it should "take" in {
+    Stream(1, 2, 3, 4).take(2).toList() shouldBe List(1, 2)
+    Stream(1, 2, 3, 4).take(1).toList() shouldBe List(1)
+    Stream(1, 2, 3, 4).take(0).toList() shouldBe Nil
+  }
```

Comment:
What happens if you try to take more elements that exist in the stream?

Reply:
I just looked at the book and I noticed they don't specify what should happen, so I figure I should probably tell you explicitly. The standard implementation of take(n) (or drop), is that if `n` is greater than the size of the Stream/collection, then behave as if `n` was equal to the size of the collection. i.e.  

```
Stream(1, 2, 3, 4).take(5).toList() shouldBe List(1, 2, 3, 4)
```